From 5aeb53dd7b5b50eddad01c684f3c8ce122d46f4c Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Wed, 28 Jul 2010 18:08:02 -0700
Subject: [PATCH 07/11] XQuartz: GLX: Don't mangle __GLXDrawable's pDraw

We were incorrectly NULLing out pDraw in __GLXDrawable instead of ours in
__GLXAquaDrawable. (we should refactor to eliminate this redundancy later)

This was causing http://xquartz.macosforge.org/trac/ticket/426
This was benign until commit f0006aa58f6cf7552a239e169ff6e7e4fda532f4
The root cause of this change was  fed7ccc481ad1caaa518cafe944c2327a5d0b6c65

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
(cherry picked from commit 98f90145d786695ecbc02a667c6ffe7c619dc67e)
---
 hw/xquartz/GL/indirect.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git xorg-server-1.8.2/hw/xquartz/GL/indirect.c xorg-server-1.8.2/hw/xquartz/GL/indirect.c
index 8092cfa..ed25c3d 100644
--- xorg-server-1.8.2/hw/xquartz/GL/indirect.c
+++ xorg-server-1.8.2/hw/xquartz/GL/indirect.c
@@ -271,8 +271,7 @@ static void __glXAquaContextDestroy(__GLXcontext *baseContext) {
 
     __GLXAquaContext *context = (__GLXAquaContext *) baseContext;
     
-    GLAQUA_DEBUG_MSG("glAquaContextDestroy (ctx 0x%x)\n",
-                     (unsigned int) baseContext);
+    GLAQUA_DEBUG_MSG("glAquaContextDestroy (ctx %p)\n", baseContext);
     if (context != NULL) {
       if (context->sid != 0 && surface_hash != NULL) {
 		lst = x_hash_table_lookup(surface_hash, x_cvt_uint_to_vptr(context->sid), NULL);
@@ -321,7 +320,7 @@ static void surface_notify(void *_arg, void *data) {
     case AppleDRISurfaceNotifyDestroyed:
         if (surface_hash != NULL)
             x_hash_table_remove(surface_hash, x_cvt_uint_to_vptr(arg->id));
-	draw->base.pDraw = NULL;
+	draw->pDraw = NULL;
 	draw->sid = 0;
         break;
 
-- 
1.7.2.1

